addons/crm/views/crm_lead_views.xml

```diff
@@ -464,6 +464,7 @@
                             'default_name': contact_name or partner_name,
                             'default_is_company': type == 'opportunity' and contact_name == False,
                             'default_company_name': type == 'opportunity' and partner_name,
+                            'default_type': 'contact',
```

Comment:
Default value is already 'contact', hence not sure it really solves something ...

Reply:
When you change the contact type from "Company" to "Individual" the type is changed to empty.

Reply:
(For history purpose) the default_get was overriding the default value of partner type to avoid the type value of the lead to be used. So specifying the default type in the fields context should indeed fix the issue.